Assets/Plugins/Source/Core/Utility/FileUtility.cs

```diff
@@ -32,27 +32,58 @@ namespace PlayEveryWare.EpicOnlineServices.Utility
     /// </summary>
     public static class FileUtility
     {
+        // Note: "\r\n" (Windows), "\n" (Unix/Linux), "\r" (older Macs)
+        private static readonly string[] LineEndings = new[] { "\r\n", "\n", "\r" };
+
         /// <summary>
         /// Generates a unique and new temporary directory inside the Temporary Cache Path as determined by Unity,
         /// and returns the fully-qualified path to the newly created directory.
         /// </summary>
         /// <returns>Fully-qualified file path to the newly generated directory.</returns>
-        public static string GenerateTempDirectory()
+        public static bool TryGetTempDirectory(out string path)
         {
             // Generate a temporary directory path.
-            string tempDirectory = Path.Combine(Application.temporaryCachePath, $"/Output-{Guid.NewGuid()}/");
+            string tempPath = Path.Combine(Application.temporaryCachePath, $"/Output-{Guid.NewGuid()}/");
 
             // If (by some crazy miracle) the directory path already exists, keep generating until there is a new one.
-            while (Directory.Exists(tempDirectory))
+            if (Directory.Exists(tempPath))
```

Comment:
Instead of opening up the possibility of an infinite loop, I limited the "retry" count to just 1. If, after that there is still a conflict, then an error is logged and the function returns false.